pulsar-broker/src/main/java/org/apache/pulsar/broker/transaction/buffer/impl/TopicTransactionBuffer.java

```diff
@@ -179,6 +179,9 @@ public void handleTxnEntry(Entry entry) {
 
                     @Override
                     public void recoverExceptionally(Exception e) {
+                        if (e instanceof PulsarClientException.BrokerMetadataException) {
```

Comment:
Other exceptions don't need to close the topic, right?

Reply:
can we add a log line here ? like "Closing topic {} due to {} while recovering the transaction buffer"